### PR TITLE
Feature/sync firestore

### PR DESCRIPTION
**Pull request by rheinardkorf**
### Description of the Change

Adding Firestore queue providers to use as alternative to SQS.

### Benefits

Can switch providers by configuration.

### Verification Process

Functional / Unit tests.

### Applicable Issues

#68 #66

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -400,6 +400,7 @@ plugins to process and writes them to a queue.
 | `SYNC_DATA` | When the database provider is set to `local` this will be where the data is stored relative to the `/srv/data` working directory. Default is `./db` |
 | `SYNC_DATABASE_DOCUMENT_PATH` |  When the database provider is set to `firestore` this value is the path to the document in Cloud Firestore. Must be in the form of `<collection>/<document>`. Default is `sync-server/wporg`. |
 | `SYNC_DATABASE_PROVIDER` | Tells the Sync Server which database provider to use; either the local file system or Google Cloud Firestore. Must be one of: `local`, `firestore`. Default is `local`. |
+| `SYNC_MESSAGE_PROVIDER` | Tells the Sync Server which message/queue provider to use; either the AWS SQS or Google Cloud Firestore. Must be one of: `aws`, `firestore`. Default is `aws`. |
```
Comment (valendesigns) [inline README.md]: We have established patterns for these config values in other environment variables so this should be one of `local`, `sqs`, or `firestore` and the default should be `local` once we integrate MongoDB. However, if we were to use `aws` then `firestore` would become `gcp`.

File: docker-compose.yml [hunk 2]
```diff
@@ -146,11 +146,14 @@ services:
             AWS_SQS_QUEUE_PHPCS: ${AWS_SQS_QUEUE_PHPCS}
             AWS_SQS_REGION: ${AWS_SQS_REGION}
             AWS_SQS_VERSION: ${AWS_SQS_VERSION}
+            FIRESTORE_QUEUE_LH: ${FIRESTORE_QUEUE_LH}
```
Comment (valendesigns) [inline docker-compose.yml]: Why would allow this to be configurable and not hard code the collection name?

File: README.md (hunk 1 above)
Comment (rheinardkorf) [reply README.md]: thats fair.... will fix in the PR

File: docker-compose.yml (hunk 2 above)
Comment (rheinardkorf) [reply docker-compose.yml]: We don't hardcode SQS.  Also, I might want to create my own Tide-based service or APP and have my collections in different places.  It does not have to be in the root of my Firestore DB.

**Pull request merged**